http/server_sent_event_stream.ts

```diff
@@ -0,0 +1,72 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+// This module is browser compatible.
+
+/**
+ * @see {@link https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events#fields}
+ */
+export interface ServerSentEventMessage {
+  /** Ignored by the client. */
+  comment?: string;
+  /** A string identifying the type of event described. */
+  event?: string;
+  /** The data field for the message. Split by new lines. */
+  data?: string;
+  /** The event ID to set the {@linkcode EventSource} object's last event ID value. */
+  id?: string | number;
+  /** The reconnection time. */
+  retry?: number;
+}
+
+/**
+ * Converts a server-sent message object into a string for the client.
+ *
+ * @see {@link https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events#event_stream_format}
+ */
+function stringify(message: ServerSentEventMessage) {
+  const lines = [];
+  if (message.comment) lines.push(`:${message.comment}`);
+  if (message.event) lines.push(`event:${message.event}`);
```

Comment:
Are newlines allowed for these? Should we validate that these fields don't contain any newlines?

Reply:
I've added a check that asserts that the single-line fields don't contain any newlines.